src/tasks/dump_db.rs

```diff
@@ -29,24 +29,57 @@ pub fn dump_db(
 /// make sure it gets deleted again even in the case of an error.
 #[derive(Debug)]
 pub struct DumpDirectory {
+    pub timestamp: chrono::DateTime<chrono::Utc>,
     pub export_dir: PathBuf,
 }
 
 impl DumpDirectory {
     pub fn create() -> Result<Self, PerformError> {
-        let timestamp = chrono::Utc::now().format("%Y-%m-%d-%H%M%S").to_string();
-        let export_dir = std::env::temp_dir().join("dump-db").join(timestamp);
+        let timestamp = chrono::Utc::now();
+        let timestamp_str = timestamp.format("%Y-%m-%d-%H%M%S").to_string();
+        let export_dir = std::env::temp_dir().join("dump-db").join(timestamp_str);
         std::fs::create_dir_all(&export_dir)?;
-        Ok(Self { export_dir })
+        Ok(Self {
+            timestamp,
+            export_dir,
+        })
     }
 
     pub fn dump_db(&self, database_url: &str) -> Result<(), PerformError> {
+        self.add_readme()?;
+        self.add_metadata()?;
         let export_script = self.export_dir.join("export.sql");
         let import_script = self.export_dir.join("import.sql");
         gen_scripts::gen_scripts(&export_script, &import_script)?;
         std::fs::create_dir(self.export_dir.join("data"))?;
         run_psql(&export_script, database_url)
     }
+
+    fn add_readme(&self) -> Result<(), PerformError> {
+        use std::io::Write;
+
+        let mut readme = File::create(self.export_dir.join("README.md"))?;
+        readme.write_all(include_bytes!("dump_db/readme_for_tarball.md"))?;
+        Ok(())
+    }
+
+    fn add_metadata(&self) -> Result<(), PerformError> {
+        #[derive(Serialize)]
+        struct Metadata<'a> {
+            timestamp: &'a chrono::DateTime<chrono::Utc>,
+            crates_io_commit: String,
+            format_version: &'static str,
+        }
+        let metadata = Metadata {
+            timestamp: &self.timestamp,
+            crates_io_commit: dotenv::var("HEROKU_SLUG_COMMIT")
+                .unwrap_or_else(|_| "unknown".to_owned()),
+            format_version: "0.1",
```

Comment:
I'm assuming you're envisioning this version number to increase its minor version with any backwards compatible schema change and increase its major version with any breaking schema change? So to give some information about compatibility with previous database dumps without needing to look up changes between git SHAs?

I'm a little concerned about the usefulness of this version number because I can very easily see us forgetting to update this with schema changes :-/ If it's not incremented meaningfully, then it's not going to be particularly useful, and people using the database dumps will need to look up what changed in the git history anyway.

Given that we recommend using the import script that does a destructive import anyway, rather than providing diffs between dumps say, is this going to provide enough value for people using the dumps to justify our maintenance of it?

Reply:
You are right, this would need a bit more specification to be useful.

I did not have in mind to increase this on every schema change, but rather everytime we change something more fundamentally to the format of the tarball, like adding a fundamentally new file.

The main goal was to allow us to make backwards-incompatible changes, and indicate these in the version number. However, this doesn't really work. If we change the dump in a way that breaks compatibility, people will notice without checking a version number, since there tool will break. And if they can't go back to the old version, this version number does not really help.

So the only way to break compatibility is to provide dumps in the new format at a different URL alongside the dumps in the old format. And that does not require a version number, so I'll remove it again.